Lib/test/test_smtplib.py

```diff
@@ -611,15 +635,18 @@ def setUp(self):
         self.sock.settimeout(15)
         self.port = support.bind_port(self.sock)
         servargs = (self.evt, self.respdata, self.sock)
-        thread = threading.Thread(target=server, args=servargs)
-        thread.start()
-        self.addCleanup(thread.join)
+        self.thread = threading.Thread(target=server, args=servargs)
+        self.thread.start()
         self.evt.wait()
         self.evt.clear()
 
     def tearDown(self):
         self.evt.wait()
         sys.stdout = self.old_stdout
+        self.thread.join()
```

Comment:
You can use support.join_thread(self.thread) to raise an error if the thread cannot be joined in 30 seconds, rather than being stuck.

Reply:
I will change then the rest of `thread.joins` of this module as there are plenty.